packages/php/email-editor/src/Engine/PersonalizationTags/class-personalization-tags-registry.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file is part of the MailPoet plugin.
+ *
+ * @package MailPoet\EmailEditor
+ */
+
+declare(strict_types = 1);
+
+namespace MailPoet\EmailEditor\Engine\PersonalizationTags;
+
+/**
+ * Registry for personalization tags.
+ */
+class Personalization_Tags_Registry {
+
+	/**
+	 * List of registered personalization tags.
+	 *
+	 * @var array
+	 */
+	private $tags = array();
+
+	/**
+	 * Initialize the personalization tags registry.
+	 *
+	 * @return void
+	 */
+	public function initialize(): void {
+		apply_filters( 'mailpoet_email_editor_register_personalization_tags', $this );
+	}
+
+	/**
+	 * Register a new personalization tag.
+	 *
+	 * @param string   $name        Unique identifier for the callback.
+	 * @param string   $tag         The tag to be used in the email content.
+	 * @param string   $category    The category of the personalization tag.
+	 * @param callable $callback    The callable function/method.
+	 * @param array    $attributes  Additional data or settings for the callback (optional).
+	 * @return void
+	 */
+	public function register( string $name, string $tag, string $category, callable $callback, array $attributes = array() ): void {
+		if ( isset( $this->tags[ $tag ] ) ) {
+			return;
```

Comment:
I considered using error_log, but it needs adding CodeSniffer ignore, which I wanted to avoid initially.
I decided to use the silent return in the first phase because if you call registration twice it could generate errors, but maybe it's correct 🤔

Reply:
I can see many approaches:

- silent return
- silently overwrite the original value
- throw an exception
- silent return with an `error_log`
- silent return with a MailPoet log

I am not saying one is the correct one, but if we want to go with one, it should probably be a conscious decision, and we should have a reason why.